javaagent/build.gradle.kts

```diff
@@ -80,28 +100,42 @@ tasks {
     }
   }
 
-  //Includes everything needed for OOTB experience
-  val shadowJar by existing(ShadowJar::class) {
-    archiveClassifier.set("all")
-    val projectsWithShadowJar = listOf(project(":instrumentation"), project(":javaagent-exporters"))
-    projectsWithShadowJar.forEach {
-      dependsOn("${it.path}:shadowJar")
+  // lightShadow is the default classifier we publish so disable the default jar.
+  jar {
+    enabled = false
+  }
+
+  val relocateJavaagentLibs by registering(ShadowJar::class) {
+    configurations = listOf(javaagentLibs)
+
+    duplicatesStrategy = DuplicatesStrategy.FAIL
+
+    archiveFileName.set("javaagentLibs-relocated.jar")
+
+    // exclude bootstrap projects from javaagent libs - they won't be added to inst/
+    dependencies {
+      exclude(project(":instrumentation-api"))
+      exclude(project(":instrumentation-api-annotation-support"))
+      exclude(project(":javaagent-bootstrap"))
+      exclude(project(":javaagent-instrumentation-api"))
     }
-    with(isolateSpec(projectsWithShadowJar))
-    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
   }
 
-  //Includes instrumentations, but not exporters
-  val lightShadow by registering(ShadowJar::class) {
-    archiveClassifier.set("")
-    dependsOn(":instrumentation:shadowJar")
-    val projectsWithShadowJar = listOf(project(":instrumentation"))
-    with(isolateSpec(projectsWithShadowJar))
+  val relocateExporterLibs by registering(ShadowJar::class) {
+    configurations = listOf(exporterLibs)
+
+    archiveFileName.set("exporterLibs-relocated.jar")
   }
 
-  withType<ShadowJar>().configureEach {
+  //Includes instrumentations, but not exporters
+  val lightShadow by registering(ShadowJar::class) {
     configurations = listOf(bootstrapLibs)
 
+    dependsOn(relocateJavaagentLibs)
+    isolateClasses(relocateJavaagentLibs.get().outputs.files)
```

Comment:
Can we just pass configuration here? Without the intermediate shadow task.

Reply:
Unfortunately not: the intermediate shadow task does the relocation, if we were to pass the configuration here we'd get not relocated files (because that'd be a separate `CopySpec` that does not share the relocation settings of the main shadow jar task).